tests/test_ahc.py

```diff
@@ -0,0 +1,57 @@
+"""Unit tests for the KNN classifier."""
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.hypersphere import Hypersphere
+from geomstats.learning.ahc import AgglomerativeHierarchicalClustering
+
+
+class TestAgglomerativeHierarchicalClustering(geomstats.tests.TestCase):
+    """Class defining the Poincare polydisk tests."""
+
+    def setUp(self):
+        """Define the parameters to test."""
+        gs.random.seed(1234)
+
+    @geomstats.tests.np_only
+    def test_fit_euclidean_distance(self):
+        """Test the 'fit' class method using the Euclidean distance."""
+        n_clusters = 2
+        dimension = 2
+        space = Euclidean(dimension=dimension)
+        distance = space.metric.dist
```

Comment:
Nice! Could you duplicate this test with a string distance argument so that the condition `if isinstance(distance, str)` is met and actually tested? We would reach 100% covering then!